src/util/irep_serialization.cpp

```diff
@@ -327,7 +327,7 @@ irep_idt irep_serializationt::read_gb_string(std::istream &in)
   char c;
   size_t length=0;
 
-  while((c = in.get()) != 0)
+  while(in.good() && (c = in.get()) != 0)
```

Comment:
Shouldn't an error be reported if such a string without 0-termination is encountered?